src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/BulkNumberSearchParameterV1RowGenerator.cs

```diff
@@ -17,14 +17,14 @@ public BulkNumberSearchParameterV1RowGenerator(SqlServerFhirModel model, SearchP
 
         internal override bool TryGenerateRow(int offset, short searchParamId, NumberSearchValue searchValue, out BulkNumberSearchParamTableTypeV1Row row)
         {
-            bool isSingleValue = searchValue.Low == searchValue.High;
+            var singleValue = searchValue.Low == searchValue.High ? searchValue.Low : null;
```

Comment:
Good question! Earlier this logic used to populate singleValue field if quantity-value is present on the resource and range(lowValue and highValue fields) when quantity-range is present on the resource. Since main reason of the PR to make lowValue and highValue field as NOT NULL(to include these in primary key constraint) so this is making a small change where we are populating the fields(lowValue and highValue) even if quantity-value is present.